src/dimension_slice.c

```diff
@@ -68,21 +68,14 @@ ts_dimension_slice_create(int dimension_id, int64 range_start, int64 range_end)
 int
 ts_dimension_slice_cmp(const DimensionSlice *left, const DimensionSlice *right)
 {
-	if (left->fd.range_start == right->fd.range_start)
-	{
-		if (left->fd.range_end == right->fd.range_end)
-			return 0;
-
-		if (left->fd.range_end > right->fd.range_end)
-			return 1;
-
-		return -1;
-	}
+	int res = (left->fd.range_start > right->fd.range_start) -
```

Comment:
Unless this makes it less convenient, maybe we could make a macro for it? Looks like this is a common pattern and used for chunk comparison as well

Reply:
Done.